pkg/util/result/result.go

```diff
@@ -66,14 +66,22 @@ func Error(msg string) *Result {
 	return FromError(err)
 }
 
-// FromError produces a Result that wraps an internal Pulumi error.
+// FromError produces a Result that wraps an internal Pulumi error.  Do not call this with a 'nil'
+// error.  A 'nil' error means that there was no problem, and in that case a 'nil' result should be
+// used instead.
 func FromError(err error) *Result {
+	if err == nil {
```

Comment:
We have `contract.Assert` for this, so you can combine this into a single call: `contract.Assert(err == nil, ...)`. No big deal, though.

Reply:
ah,  I don't mind this :)